spring-web-app/src/test/java/com/epam/viktoryia/springdatabase/ClientRestTemplateTest.java

```diff
@@ -0,0 +1,57 @@
+package com.epam.viktoryia.springdatabase;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.*;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ClientRestTemplateTest extends Mockito {
+
+    private static final Integer ID = 1;
+    private static final String URL = "http://localhost:8080/employee/";
+
+    @Mock
+    RestTemplate restTemplate;
+    @InjectMocks
+    ClientRestTemplate clientRestTemplate;
+    @Captor
+    ArgumentCaptor <HttpEntity> httpEntityCaptor = ArgumentCaptor.forClass(HttpEntity.class);
+
+    @Test
+    public void testDoPost() throws Exception {
+
+        clientRestTemplate.doPost();
+        verify(restTemplate).postForObject(Mockito.eq(URL), httpEntityCaptor.capture(), Mockito.eq(List.class));
```

Comment:
Regarding to a captor usage look at https://www.journaldev.com/21892/mockito-argumentcaptor-captor-annotation, please, for example